internal/services/authorizer/internal_test.go

```diff
@@ -0,0 +1,99 @@
+package authorizer
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/credentials"
+)
+
+func TestPeerCredsInfoAuthType(t *testing.T) {
+	t.Parallel()
+
+	p := peerCredsInfo{
+		uid: 11111,
+		pid: 22222,
+	}
+
+	require.Equal(t, "uid: 11111, pid: 22222", p.AuthType(), "AuthType returns expected uid and pid")
+}
+func TestServerPeerCredsHandshake(t *testing.T) {
+	t.Parallel()
+
+	s := serverPeerCreds{}
+
+	socket := filepath.Join(t.TempDir(), "authd.sock")
+	l, err := net.Listen("unix", socket)
+	if err != nil {
+		t.Fatalf("Couldn't listen on socket: %v", err)
+	}
+	defer l.Close()
```

Comment:
I've found this not being called on some failures, so I'd use `t.Cleanup(l.Close)` instead.

Or a more complex function with `require.NoError` to mimic the final behavior here.

Reply:
Well, I mostly noticed in cases where had to remove things, and if done during a test defer it didn't happen on failures... I never went through checking the details, but I defeinitely saw it not happening 100%.

Reply:
t.Cleanup is exactly the same mechanism on the compiler than defer, so I doubt one will run when the other didn’t. It will be great to see why.

The only case I can think of (but Cleanup will not help there either) is `os.Exit()`, which created some issues in the pass in `TestMain`: https://blog.englund.nu/golang,/testing/2017/03/12/using-defer-in-testmain.html but it’s not the case anymore since https://github.com/golang/go/issues/34129 is fixed.